azurelinuxagent/common/logger.py

```diff
@@ -125,18 +126,47 @@ def write(self, level, msg):
 
 
 class FileAppender(object):
-    def __init__(self, level, path):
+    def __init__(self, level, path, max_bytes, backup_count):
         self.level = level
         self.path = path
+        self.max_bytes = max_bytes
+        self.backup_count = backup_count
 
     def write(self, level, msg):
         if self.level <= level:
+
+            if self.shouldRollover(msg):
+                self.doRollover()
+
             try:
                 with open(self.path, "a+") as log_file:
                     log_file.write(msg)
             except IOError:
                 pass
 
+    def shouldRollover(self, msg):
+        if self.backup_count == 0:
+            return False
+
+        if os.stat(self.path).st_size + len(msg) > self.max_bytes:
+            return True
+
+        return False
+
+    def doRollover(self):
+        if self.backup_count > 0:
+            for i in range(self.backup_count-1, 0, -1):
+                src = "%s.%d" % (self.path, i)
+                dst = "%s.%d" % (self.path, i + 1)
+                if os.path.exists(src):
+                    if os.path.exists(dst):
+                        os.remove(dst)
+                    os.rename(src, dst)
```

Comment:
Why not compress the backup?

Reply:
For my use case I don't need compression, old log should be readable in plain text.